grr/server/grr_response_server/bin/grrafana_test.py

```diff
@@ -198,7 +202,25 @@ def testQuery(self):
           "datapoints": [[0.31883034110069275, 1597328417823],
                          [0.07246342301368713, 1597328419403]]
       }])
-      self.assertRaises(NameError, self.client.post, "/query", json=_TEST_INVALID_TARGET_QUERY)
+
+  def testQueryInvalidRequest(self):
+    conn = _MockConnReturningRecords([
+        _TEST_CLIENT_RESOURCE_USAGE_RECORD_1,
+        _TEST_CLIENT_RESOURCE_USAGE_RECORD_2
+    ])
+    with mock.patch.object(fleetspeak_connector, "CONN", conn):
+      self.assertRaises(NameError,
+                        self.client.post,
+                        "/query",
+                        json=_TEST_INVALID_TARGET_QUERY)
+
+  def testGrafanaTimestampToTimestampObj(self):
```

Comment:
This is a standalone function that is used by Grrafana HTTP server, but has no direct relation to it. The current class (`GrrafanaTest`) is a set of tests for the HTTP server. I'd put this test in a separate test suite (call it GrafanaTimestampToTimestampObjTest).